cpp/src/bitmask/bitmask_ops.cu

```diff
@@ -93,3 +81,20 @@ gdf_error apply_bitmask_to_bitmask(gdf_size_type & out_null_count, gdf_valid_typ
 	return GDF_SUCCESS;
 }
 
+gdf_error apply_bitmask_to_bitmask(gdf_size_type & out_null_count, gdf_valid_type * valid_out, gdf_valid_type * valid_left, gdf_valid_type * valid_right,
+		cudaStream_t stream, gdf_size_type num_values){
+
+	gdf_size_type num_chars_bitmask = ( ( num_values +( GDF_VALID_BITSIZE - 1)) / GDF_VALID_BITSIZE );
+	thrust::device_ptr<gdf_valid_type> valid_out_ptr = thrust::device_pointer_cast(valid_out);
```

Comment:
I know that you didn't write it this way, but since you're already updating it... there's no need for all this verbose `thrust::iterator` and `thrust::device_ptr` stuff is just messy. You can just pass the `valid_out` and `valid_left` pointers directly into the `thrust::transform` call.

You're already explicitly telling it to use the GPU in the `rmm::exec_policy`, so there's no need to use the `thrust::device_ptr`.

Reply:
I had created this issue before I saw you had already started updating this code: https://github.com/rapidsai/cudf/issues/1057